From 6cf996111005b606cad163cbcbaa4da2d053de38 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Corentin=20No=C3=ABl?= <corentin@elementary.io>
Date: Wed, 28 Oct 2015 16:17:19 +0100
Subject: [PATCH] Fixed VAPI compilation.

---
 contrib/vapi/CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/contrib/vapi/CMakeLists.txt b/contrib/vapi/CMakeLists.txt
index 393aef1..223ef39 100644
--- a/contrib/vapi/CMakeLists.txt
+++ b/contrib/vapi/CMakeLists.txt
@@ -15,7 +15,7 @@ add_custom_command (OUTPUT ${APPSTREAM_VAPI_FILE}
 					--pkg glib-2.0 --pkg gio-2.0 --pkg gobject-2.0
 			WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}
 			DEPENDS appstream
-					${CMAKE_BINARY_DIR}/src/Appstream-${AS_VERSION_BASE}.typelib
+					${CMAKE_BINARY_DIR}/src/AppStream-${AS_VERSION_BASE}.typelib
 			VERBATIM
 )
 add_custom_target(vapi ALL DEPENDS ${APPSTREAM_VAPI_FILE})
