Index: appstream/src/as-component.c
===================================================================
--- appstream.orig/src/as-component.c
+++ appstream/src/as-component.c
@@ -1709,7 +1709,7 @@ as_component_add_tag (AsComponent *cpt,
 	g_autofree gchar *tag_full = g_strconcat (ns, "::", tag, NULL);
 
 	/* sanity check */
-	if (g_strstr_len (tag, -1, "::") != NULL)
+	if (as_is_empty (tag_full) || g_strstr_len (tag, -1, "::") != NULL)
 		return FALSE;
 
 	for (guint i = 0; i < priv->tags->len; i++) {
@@ -3071,9 +3071,9 @@ as_component_get_value_flags (AsComponen
 static gboolean
 as_component_has_desktop_group (AsComponent *cpt, const gchar *desktop_group)
 {
-	guint i;
+	if (as_is_empty (desktop_group)) return TRUE;
 	g_auto(GStrv) split = g_strsplit (desktop_group, "::", -1);
-	for (i = 0; split[i] != NULL; i++) {
+	for (guint i = 0; split[i] != NULL; i++) {
 		if (!as_component_has_category (cpt, split[i]))
 			return FALSE;
 	}
@@ -4649,7 +4649,9 @@ as_component_to_xml_node (AsComponent *c
 		for (guint i = 0; i < priv->tags->len; i++) {
 			xmlNode *tag_node = NULL;
 			g_auto(GStrv) parts = g_strsplit (g_ptr_array_index (priv->tags, i), "::", 2);
-			g_assert (parts[1] != NULL);
+			if (as_is_empty (parts[0]) || as_is_empty (parts[1])) {
+				continue;
+			}
 			tag_node = as_xml_add_text_node (tags_node, "tag", parts[1]);
 			as_xml_add_text_prop (tag_node, "namespace", parts[0]);
 		}
@@ -5786,7 +5788,9 @@ as_component_emit_yaml (AsComponent *cpt
 
 		for (guint i = 0; i < priv->tags->len; i++) {
 			g_auto(GStrv) parts = g_strsplit (g_ptr_array_index (priv->tags, i), "::", 2);
-			g_assert (parts[1] != NULL);
+			if (as_is_empty (parts[0]) || as_is_empty (parts[1])) {
+				continue;
+			}
 
 			as_yaml_mapping_start (emitter);
 			as_yaml_emit_entry (emitter, "namespace", parts[0]);
